extras/vagrantExample/src/main/vagrant/Vagrantfile

```diff
@@ -92,67 +100,106 @@ EOF
-      curl -f "$@"
+      ### curl --fail treat http status >= 400 as an error. --location follow redirects status>=300
+      curl --silent --show-error --fail --location "$@"
       if [ $? -ne 0 ]; then
-        echo "--------------------------"
-        echo "-"
-        echo "- download failed" "$@"
-        echo "-"
-        echo "-"   exiting ...
-        echo "-"
-        echo "--------------------------"
-        exit 1
+        echoerr "--------------------------"
+        echoerr "-"
+        echoerr "- download failed" "$@"
+        echoerr "-"
+        echoerr "-"   exiting ...
+        echoerr "-"
+        echoerr "--------------------------"
+        exit 800
       fi
     }
+    ### wait for a directory to exist or 60 seconds timeout
+    function waitForDeploy {
+        waitfordir="$1"
+        timeout=60
+        while [[ ! -d  "$waitfordir" ]]  
+        do
+            sleep 5
+            let timeout-=5
+            if [[ $timeout -le "0" ]]; then 
+                echo "Timeout waiting for war to deploy, $waitfordir still does not exist."; 
+                exit 401 
+            fi
+        done
+    }
     
     echo "- Hadoop"
-    download -SLO http://apache.mirrors.tds.net/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz 
-    tar -zxf hadoop-${HADOOP_VERSION}.tar.gz
+    hadoopUrl=http://apache.mirrors.tds.net/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz
+    if [[ ! -d ${HADOOP_HOME} ]] ; then 
+        echo "Downloading $hadoopUrl"
+        download $hadoopUrl \
+        | tar -zxC /home/vagrant || exit 101
+    fi
 
     echo "- Zookeeper"
-    download -SLO http://archive-primary.cloudera.com/cdh4/cdh/4/zookeeper-${ZOOKEEPER_VERSION}.tar.gz
-    tar -zxf zookeeper-${ZOOKEEPER_VERSION}.tar.gz
-    
+    zookeeperUrl=http://archive-primary.cloudera.com/cdh4/cdh/4/zookeeper-${ZOOKEEPER_VERSION}.tar.gz
+    if [[ ! -d ${ZOOKEEPER_HOME} ]] ; then
+        echo "Downloading $zookeeperUrl"
+        download $zookeeperUrl \
+        | tar -zxC /home/vagrant || exit 102
+    fi
+
     echo "- Accumulo"
-    download -SLO http://apache.mirrors.pair.com/accumulo/${ACCUMULO_VERSION}/accumulo-${ACCUMULO_VERSION}-bin.tar.gz
-    tar -zxf accumulo-${ACCUMULO_VERSION}-bin.tar.gz
+    accumuloUrl=http://apache.mirrors.pair.com/accumulo/${ACCUMULO_VERSION}/accumulo-${ACCUMULO_VERSION}-bin.tar.gz
+	if [[ ! -d ${ACCUMULO_HOME} ]] ; then
+		echo "Downloading $accumuloUrl"
+		download $accumuloUrl \
+		| tar -zxC /home/vagrant || exit 103
+	fi
     
     echo "Configuring Zookeeper..."
-    sudo mkdir /var/zookeeper
+    sudo mkdir --parents /var/zookeeper
     sudo chown vagrant:vagrant /var/zookeeper
 
     echo "Running Zookeeper..."
-    cp zookeeper-${ZOOKEEPER_VERSION}/conf/zoo_sample.cfg zookeeper-${ZOOKEEPER_VERSION}/conf/zoo.cfg
-    sudo zookeeper-${ZOOKEEPER_VERSION}/bin/zkServer.sh start
+    cp ${ZOOKEEPER_HOME}/conf/zoo_sample.cfg ${ZOOKEEPER_HOME}/conf/zoo.cfg
```

Comment:
because dataDir in zoo.cfg is set to a /tmp/zookeeper dir, we have problems with Accumulo initialization after a restart (data is basically lost). Could you change zoo.cfg to have dataDir=/var/zookeeper for example?